crates/engine/tree/src/persistence.rs

```diff
@@ -323,19 +321,21 @@ mod tests {
         Persistence::spawn_new(provider, static_file_handle, pruner)
     }
 
-    #[tokio::test]
-    async fn test_save_blocks_empty() {
+    #[test]
+    fn test_save_blocks_empty() {
         let persistence_handle = default_persistence_handle();
 
         let blocks = vec![];
+        let (tx, rx) = oneshot::channel();
 
-        let hash = persistence_handle.save_blocks(blocks).await;
+        persistence_handle.save_blocks(blocks, tx);
```

Comment:
the async functions are quite useful for testing, so I think we want to keep them (for now)

here we could use the send_action directly

https://github.com/paradigmxyz/reth/blob/a1e93b79aa01dd57108c6b326d781140b13a49db/crates/engine/tree/src/persistence.rs#L244-L251

Reply:
ok, moved back to async, but do we need to use `send_action`? or is it for exercising that method?